### PR TITLE
CHI-1735: fix clear form, mapping tweaks

### PR DESCRIPTION
**Pull request by stephenhand**
Primary reviewer:

## Description

* Fixes 'clear form' for resources search
* Adds full descriptions of all filters being used on results page
* Fixes some mappings and makes conversion mapping code more strongly typed
* Adds full list of Canadian Cities as provided by KHP

We will do full translations once we are getting ready to launch & UIs are finalised

### Checklist
- [X] Corresponding issue has been opened
- [X] New tests added
- N/A Feature flags added
- N/A Strings are localized
- [X] Tested for chat contacts
- [X] Tested for call contacts

### Related Issues
Fixes CHI-1735

### Verification steps

* Deploy to Aselo Development
* Ensure cities appear when provinces are selected (we don't seem to have any for Quebec, checking that)
* Try various selections and ensure the descriptions on the results page reflect the selections
* Experiment with filters and see if they return sensible results

### CODE REVIEW

File: plugin-hrm-form/src/components/resources/convertKHPResourceAttributes.ts [hunk 1]
```diff
@@ -15,35 +15,40 @@
...
-  const propVal = (attributes ?? {})[keyName];
-  if (!propVal) {
-    return '';
-  }
-  const propValueByLanguage = propVal.find(item => item?.language === language || item?.language === '');
-  if (propValueByLanguage && 'value' in propValueByLanguage && typeof propValueByLanguage.value === 'string') {
-    return propValueByLanguage.value;
-  } else if (propVal && 'value' in propVal[0] && typeof propVal[0].value === 'boolean') {
-    // For keysToKeep, do not change to Yes / No responses
-    const keysToKeep = ['primaryLocationIsPrivate', 'isLocationPrivate', 'isPrivate'];
```
Comment (mythilytm) [inline plugin-hrm-form/src/components/resources/convertKHPResourceAttributes.ts:33]: > const keysToKeep = ['primaryLocationIsPrivate', 'isLocationPrivate', 'isPrivate'];

How are these mapped / used now?

File: plugin-hrm-form/src/components/resources/convertKHPResourceAttributes.ts [hunk 2]
```diff
@@ -239,8 +260,8 @@ export const convertKHPResourceAttributes = (
     howIsServiceOffered: getAttributeValuesAsCsv(attributes, language, 'howIsServiceOffered'),
     accessibility: getAttributeValue(attributes, language, 'accessibility'),
     documentsRequired: extractRequiredDocuments(attributes.documentsRequired, language),
-    primaryLocationIsPrivate: getAttributeValue(attributes, language, 'primaryLocationIsPrivate'),
+    primaryLocationIsPrivate: getBooleanAttributeValue(attributes, 'primaryLocationIsPrivate'),
```
Comment (mythilytm) [inline plugin-hrm-form/src/components/resources/convertKHPResourceAttributes.ts:263]: I see

File: plugin-hrm-form/src/components/resources/convertKHPResourceAttributes.ts [hunk 3]
```diff
@@ -15,35 +15,40 @@
...
-    } else if (propVal[0].value === false && !keysToKeep.includes(keyName)) {
-      return 'No';
     }
+    return 'No';
   }
-  return propVal[0].value;
+  return (value ?? '').toString();
+};
+
+const getBooleanAttributeValue = (attributes: Attributes | undefined, keyName: string) => {
```
Comment (mythilytm) [inline plugin-hrm-form/src/components/resources/convertKHPResourceAttributes.ts:44]: 👍

File: plugin-hrm-form/src/components/resources/convertKHPResourceAttributes.ts (hunk 2 above)
Comment (stephenhand) [reply plugin-hrm-form/src/components/resources/convertKHPResourceAttributes.ts:263]: Once I put the types in it was easier to do it as a separate function, TS complained less ;-)

**Review by mythilytm — APPROVED** [review]
Works great

**Pull request merged**